### PR TITLE
[cssom-view][geometry] Test re-introduction of DOMRectList

### PR DESCRIPTION
**Pull request by zcorpan**
See https://github.com/w3c/csswg-drafts/issues/1479

idlharness.js currently does not test anything for interfaces
marked with [NoInterfaceObject], so this tests some IDL aspects
in css/geometry-1/DOMRectList.html instead.

Also test that ClientRect and ClientRectList are not supported.

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/245277469)

# Firefox (nightly) #
Testing web-platform-tests at revision 880b3ee7b693af4726452541ac958369c292ed16
Using browser at version BuildID 20170612100241; SourceStamp 27cad9749cddf68e11fdd4e5d73dad84a8f8cf23
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>3 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6299/css/geometry-1/DOMRectList.html">/css/geometry-1/DOMRectList.html</a></summary>


|              Subtest              | Results |                 Messages                |
|-----------------------------------|---------|-----------------------------------------|
|                                   | OK      |                                         |
| `DOMRectList [NoInterfaceObject]` | FAIL    | `assert_false: expected false got true` |
| `DOMRectList [LegacyArrayClass]`  | PASS    |                                         |
| `DOMRectList length`              | PASS    |                                         |
| `DOMRectList indexed getter`      | PASS    |                                         |
| `DOMRectList item()`              | PASS    |                                         |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/6299/cssom-view/DOMRectList.html">/cssom-view/DOMRectList.html</a></summary>


|          Subtest           | Results | Messages |
|----------------------------|---------|----------|
|                            | OK      |          |
| `Element getClientRects()` | PASS    |          |
| `Range getClientRects()`   | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/6299/cssom-view/historical.html">/cssom-view/historical.html</a></summary>


|           Subtest            | Results | Messages |
|------------------------------|---------|----------|
|                              | OK      |          |
| `Support for ClientRectList` | PASS    |          |
| `Support for ClientRect`     | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/245277470)

# Sauce (safari) #
Testing web-platform-tests at revision 880b3ee7b693af4726452541ac958369c292ed16
Using browser at version 10.0
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>3 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6299/css/geometry-1/DOMRectList.html">/css/geometry-1/DOMRectList.html</a></summary>


|              Subtest              | Results |                                            Messages                                            |
|-----------------------------------|---------|------------------------------------------------------------------------------------------------|
|                                   | OK      |                                                                                                |
| `DOMRectList [NoInterfaceObject]` | PASS    |                                                                                                |
| `DOMRectList [LegacyArrayClass]`  | FAIL    | `assert_true: expected true got false`                                                         |
| `DOMRectList length`              | PASS    |                                                                                                |
| `DOMRectList indexed getter`      | FAIL    | `assert_equals: domRectList[0] expected "[object DOMRect]" but got "[object ClientRect]"`      |
| `DOMRectList item()`              | FAIL    | `assert_equals: domRectList.item(0) expected "[object DOMRect]" but got "[object ClientRect]"` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/6299/cssom-view/DOMRectList.html">/cssom-view/DOMRectList.html</a></summary>


|          Subtest           | Results |                                      Messages                                      |
|----------------------------|---------|------------------------------------------------------------------------------------|
|                            | OK      |                                                                                    |
| `Element getClientRects()` | FAIL    | `assert_equals: expected "[object DOMRectList]" but got "[object ClientRectList]"` |
| `Range getClientRects()`   | FAIL    | `assert_equals: expected "[object DOMRectList]" but got "[object ClientRectList]"` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/6299/cssom-view/historical.html">/cssom-view/historical.html</a></summary>


|           Subtest            | Results |                 Messages                |
|------------------------------|---------|-----------------------------------------|
|                              | OK      |                                         |
| `Support for ClientRectList` | FAIL    | `assert_false: expected false got true` |
| `Support for ClientRect`     | FAIL    | `assert_false: expected false got true` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/245277476)

# Chrome (unstable) #
Testing web-platform-tests at revision 880b3ee7b693af4726452541ac958369c292ed16

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/245277477)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision 880b3ee7b693af4726452541ac958369c292ed16

**Comment by domenic** [comment]
> idlharness.js currently does not test anything for interfaces marked with [NoInterfaceObject], so this tests some IDL aspects in css/geometry-1/DOMRectList.html instead.

Can't you tel it how to create an object, and then it will run idlharness tests on it? I'm not familiar with how to do so myself but I think I've seen similar code in other tests.

**Comment by zcorpan** [comment]
Yes, done. Though idlharness could in theory test more for NoInterfaceObject, there are some TODOs in the code (like, for example, checking that the interface object is not exposed).

File: css/geometry-1/structured-serialization.html [hunk 1]
```diff
@@ -156,4 +157,9 @@
       throw new Error(`Test bug: ${constr} has no test for non-initial values`);
   }
 });
+
+promise_test((t) => {
```
Comment (foolip) [inline css/geometry-1/structured-serialization.html]: Using promise_rejects here would make the test smaller.

**Review by foolip — APPROVED** [review]
Just one nit.

File: css/geometry-1/structured-serialization.html (hunk 1 above)
Comment (foolip) [inline css/geometry-1/structured-serialization.html]: Fixed myself, will merge when Travis is happy.

**Pull request merged**